### DISCUSSION TITLE
XFS with crc=1 and reflink=1

### DISCUSSION THREAD

**Discussion by hashworks** [minio/minio#17978] [Q&A] [answered]
MinIO requests XFS as the FS for the JBOD storage pool. For a 15T disk the default `mkfs.xfs` reserves about 104G for metadata:
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda1        15T  104G   15T   1% /mnt/minio1
```

This is caused by the default settings `crc=1` (`finobt=1`) and `reflink=1`, which can be viewed by calling `xfs_info /dev/sda1`:
```
meta-data=/dev/sda1              isize=512    agcount=15, agsize=268435455 blks
         =                       sectsz=4096  attr=2, projid32bit=1
         =                       crc=1        finobt=1, sparse=1, rmapbt=0
         =                       reflink=1    bigtime=1 inobtcount=1 nrext64=0
data     =                       bsize=4096   blocks=3906444027, imaxpct=5
         =                       sunit=0      swidth=0 blks
naming   =version 2              bsize=4096   ascii-ci=0, ftype=1
log      =internal log           bsize=4096   blocks=521728, version=2
         =                       sectsz=4096  sunit=1 blks, lazy-count=1
realtime =none                   extsz=4096   blocks=0, rtextents=0
```

Does MinIO actually need those settings? Or should one format the disks with `crc=0` and `reflink=0`?

**Comment by harshavardhana** [comment, marked as answer]
That is up-to you, we recommend default XFS. We have no idea what are the implications of these being turned off.

**Comment by hashworks** [comment]
I've looked into this myself.

It seems setting `crc=0` breaks various features one might need, f.e. 2038 support, so I wouldn't recommend it: https://www.reddit.com/r/filesystems/comments/tl2mhw/how_dangerous_is_to_format_a_hard_drive_to_xfs/

However, `reflink=0` seems perfectly reasonable since it is mainly used for deduplication, which AFAIK minio does by itself: https://blogs.oracle.com/linux/post/xfs-data-block-sharing-reflink

Formatting a 15T drive with `mkfs.xfs -m reflink=0` saves 90G.

**Reply by allahshukur-ahmadzada** [reply]
Hi @hashworks , I am planning to do the same (reflink=0,crc=1), later on did you have any issues or changed your mind?

**Reply by hashworks** [reply]
No issues with `reflink=0` over here.
